File: Csharp/Lab8/Lab8/Lab8/cng/Model/Entity/Container/Tree.cs
using System;
using System.Text;
using Lab8.cng.Model.Entity.Abstracts;

namespace Lab8.cng.Model.Entity.Container
{
    public class Tree
    {
        private Toy[] _toysMass;
        private static Tree _tree = null;

        public static Tree GetInstanceOfTree(ToyType toySType)
        {
            if (null == _tree)
            {
                _tree = new Tree();
                if (toySType==ToyType.Garland)
                {
                    _tree.Add(new Garland());
                }
                else if (toySType==ToyType.Lights)
                {
                    _tree.Add(new Lights());
                }
                else if (toySType==ToyType.Stuffedtoys)
                {
                    _tree.Add(new StuffedToys());
                }
            }

            return _tree;
        }

        private Tree()
        {
        }

        private Tree(Toy []toys)
        {
            //mass
            this._toysMass = toys;
        }


        public Toy[] GetToysMass()
        {
            return _toysMass;
        }

        public int GetSize()
        {
            return _toysMass.Length;
        }

        public Toy GetToy(int index)
        {
            return _toysMass[index];
        }

//priceTree
        public void ChangeToy(Toy newToy, int index)
        {
            _toysMass[index] = newToy;
        }

        public void SetToy(Toy newToy, int index)
        {
            Toy[] temp = new Toy[_toysMass.Length + 1];
            for (int i = 0; i < _toysMass.Length; i++)
            {
                if (i < index)
                {
                    temp[i] = _toysMass[i];
                }
                else
                {
                    temp[i + 1] = _toysMass[i];
                }
            }

            temp[index] = newToy;
            _toysMass = temp;
        }

        public double GetPrice(int index)
        {
            return _toysMass[index].GetPrice();
        }

        public void Add(Toy newToy)
        {
            try
            {
                int index = _toysMass.Length;
                Toy[] temp = new Toy[_toysMass.Length + 1];
                for (int i = 0; i < _toysMass.Length; i++)
                {
                    if (i < index)
                    {
                        temp[i] = _toysMass[i];
                    }
                    else
                    {
                        temp[i + 1] = _toysMass[i];
                    }
                }

                temp[index] = newToy;
                _toysMass = temp;
            }
            catch (Exception)
            {
                _toysMass = new Toy[1];
                _toysMass[0] = newToy;
            }
        }

        

        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();
            foreach (Toy toy in _toysMass) {
                sb.Append(toy).Append("\n");
            }
            return sb.ToString();
        }
    }

    public enum ToyType
    {
         Garland, Lights, Stuffedtoys   
    }
    
}

File: Java/Java EE/Servlets/FirstServlet/src/Model/entity/container/Tree.java
package Model.entity.container;


import Model.entity.abstracts.Toy;

public class Tree {
//        private ArrayList<Toy> toysList;
//    public Tree(Toy...toys){
//        this.toysList.addAll(Arrays.asList(toys));
//    }
    private Toy[] toysMass;

    public Tree() {

    }

    public Tree(Toy... toys) {//mass
        this.toysMass = toys;
    }

    public Toy[] getToysMass() {
        return toysMass;
    }

    public int getSize(){
        return toysMass.length;
    }

    public Toy getToy(int index){
        return toysMass[index];
    }
//priceTree
    public void changeToy(Toy newToy, int index){
        toysMass[index] = newToy;
    }

    public void setToy(Toy newToy, int index){
        Toy[] temp = new Toy[toysMass.length + 1];
        for (int i = 0; i < toysMass.length; i++) {
            if (i < index) {
                temp[i] = toysMass[i];
            } else {
                temp[i + 1] = toysMass[i];
            }
        }
        temp[index] = newToy;
        toysMass = temp;
    }

    public double getPrice(int index){
        return toysMass[index].getPrice();
    }

    public void add(Toy newToy) {
        try{
            int index = toysMass.length;
            Toy[] temp = new Toy[toysMass.length + 1];
            for (int i = 0; i < toysMass.length; i++) {
                if (i < index) {
                    temp[i] = toysMass[i];
                } else {
                    temp[i + 1] = toysMass[i];
                }
            }
            temp[index] = newToy;
            toysMass = temp;
        } catch (NullPointerException e){
            toysMass = new Toy[1];
            toysMass[0] = newToy;
        }
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        for (Toy toysMas : this.toysMass) {
            sb.append(toysMas.toString()).append("\n");
        }
        return sb.toString();
    }
}
